File: src/main/scala/org/openapitools/client/api/CdashigProduct.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import CdashigProduct._

case class CdashigProduct (
  name: Option[String],
label: Option[String],
description: Option[String],
source: Option[String],
effectiveDate: Option[String],
registrationStatus: Option[String],
version: Option[String],
links: Option[CdashigProductLinks],
classes: Option[List[CdashigClass]])

object CdashigProduct {
  import DateTimeCodecs._

  implicit val CdashigProductCodecJson: CodecJson[CdashigProduct] = CodecJson.derive[CdashigProduct]
  implicit val CdashigProductDecoder: EntityDecoder[CdashigProduct] = jsonOf[CdashigProduct]
  implicit val CdashigProductEncoder: EntityEncoder[CdashigProduct] = jsonEncoderOf[CdashigProduct]
}

File: src/cdisc_library_api/specs/cdashig_product.clj
(ns cdisc-library-api.specs.cdashig-product
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [cdisc-library-api.specs.cdashig-product-links :refer :all]
            [cdisc-library-api.specs.cdashig-class :refer :all]
            )
  (:import (java.io File)))


(def cdashig-product-data
  {
   (ds/opt :name) string?
   (ds/opt :label) string?
   (ds/opt :description) string?
   (ds/opt :source) string?
   (ds/opt :effectiveDate) string?
   (ds/opt :registrationStatus) string?
   (ds/opt :version) string?
   (ds/opt :_links) cdashig-product-links-spec
   (ds/opt :classes) (s/coll-of cdashig-class-spec)
   })

(def cdashig-product-spec
  (ds/spec
    {:name ::cdashig-product
     :spec cdashig-product-data}))
